src/scheduler.rs

```diff
@@ -434,6 +434,82 @@ impl Scheduler {
+    /// communicate over secure connections in production.
+    pub async fn send_payjoin<'a>(
+        &self,
+        uri: bip78::Uri<'_>,
+        danger_accept_self_signed_uri: bool,
+    ) -> Result<Txid, SchedulerError> {
+        log::debug!("get original_psbt");
+        let pj_uri = bip78::UriExt::check_pj_supported(uri)
+            .map_err(|e| SchedulerError::UriDoesNotSupportPayJoin(e.to_string()))?;
+        let (original_psbt, leased_utxos) = if let Some(amount) = pj_uri.amount {
+            log::debug!("funding original_psbt");
+            self.lnd.fund_original_psbt(&pj_uri.address, amount).await?
+        } else {
+            return Err(SchedulerError::UriDoesNotSupportPayJoin(
+                "Missing amount. Make sure the bip21 uri specifies an amount".to_string(),
+            ));
+        };
+        log::debug!("sign original_psbt");
+        let original_psbt = self.lnd.sign_psbt(original_psbt).await?;
+        log::debug!("request_payjoin");
+        let res = self.request_payjoin(pj_uri, original_psbt, danger_accept_self_signed_uri).await;
+        if res.is_err() {
+            log::debug!("release_utxos");
+            self.lnd.release_utxos(leased_utxos).await?;
+        } // else, those utxos are now spent and don't need to be released
+        res
+    }
+
+    async fn request_payjoin(
+        &self,
+        pj_uri: bip78::PjUri<'_>,
+        original_psbt: PartiallySignedTransaction,
+        danger_accept_self_signed_uri: bool,
+    ) -> Result<Txid, SchedulerError> {
+        use bip78::PjUriExt;
+
+        let pj_params = bip78::sender::Configuration::with_fee_contribution(
+            bip78::bitcoin::Amount::from_sat(10000),
+            None,
+        );
+        let (req, ctx) = pj_uri
+            .create_pj_request(original_psbt, pj_params)
+            .map_err(|_| SchedulerError::Internal("failed to make http pj request"))?;
+
+        let http = reqwest::ClientBuilder::new()
+            .danger_accept_invalid_certs(danger_accept_self_signed_uri)
+            .build()
+            .map_err(|_| SchedulerError::Internal("Failed to build http client"))?;
+
+        let response = http
+            .post(req.url)
+            .header("Content-Type", "text/plain")
+            .body(reqwest::Body::from(req.body))
+            .send()
+            .await
+            .map_err(|_| SchedulerError::Internal("PayJoin http request failed"))?;
+
+        log::debug!("res: {:#?}", &response);
+        let response =
+            response.bytes().await.map_err(|_| SchedulerError::Internal("Bad response"))?;
+        let payjoin_psbt = ctx
+            .process_response(response.to_vec().as_slice())
+            .map_err(|_| SchedulerError::Internal("bip78::sender ValidationError"))?;
+
+        log::debug!("Proposed psbt: {:#?}", &payjoin_psbt);
+        let tx = self.lnd.finalize_psbt(payjoin_psbt).await?;
+        log::debug!("broadcasting");
+        let txid = self.lnd.broadcast(tx).await?;
```

Comment:
Is the `?` error different to our `LndError::Publish` error? Could maybe `unwrap_or_else`, but not important

Reply:
`?` calls `from(e: LndError) -> SchedulerError` resulting in `SchedulerError::PayJoinError(LndError::Publish(...))` so that this bubbles up to the front end.

How could `unwrap_or_else` be used here?